### PR TITLE
reparse individual non-parsed words after full sentence

### PR DESCRIPTION
**Pull request by unhammer**
as noted in https://github.com/apertium/apertium-recursive/issues/80#issuecomment-1691484763 this tries to do a reparse of single, unparsed words after full parse. 

It seems to fix things in corpus runs. 

Could it break anything @mr-martian ?

### CODE REVIEW

File: src/rtx_processor.cc [hunk 1]
```diff
@@ -1798,6 +1798,32 @@ RTXProcessor::processGLR(UFILE *out)
...
+          temp->init(mx, ch);
+          temp->id = ++newBranchId;
+          temp->stringVars = variables;
+          temp->wblankVars = wblank_variables;
+          temp->chunkVars = vector<Chunk*>(varCount, NULL);
+          checkForReduce(parseGraph, temp);
+
+          list<Chunk*> outputQueueReparsed;
+          parseGraph[0]->getChunks(outputQueueReparsed, parseGraph[0]->length-1);
+          for(Chunk* ch2 : outputQueueReparsed) { // should be at most one
```
Comment (mr-martian) [inline src/rtx_processor.cc:1820]: Did I really use a `std::list` for something? ew.

Also, there actually are pathological situations in which a reduction rule could match 1 LU and output multiple chunks.

**Comment by unhammer** [comment]
ok so it breaks tests :) Is it an actual failure? a tag changed from pl to sg

**Comment by mr-martian** [comment]
In the first test that fails, there are 2 rules and they both change the same variable, and I think something might be going wrong with the ordering of those operations.

File: src/rtx_processor.cc (hunk 1 above)
Comment (unhammer) [reply src/rtx_processor.cc:1820]: bddbfb755c262ab4c24df4cdf76ebbe05c8d6b0c is this how we splice-and-replace, or is there a nicer way?

**Comment by unhammer** [comment]
OK so it seems all the tests run in `processGLR`, I thought xmls used `processTRX`.

**Review by mr-martian — APPROVED** [review]

**Pull request merged**